tools/tesseratos/src/tesseratos/world_inspector/plugin.cpp

```diff
@@ -20,8 +24,112 @@ using cubos::engine::Entity;
 using cubos::engine::Opt;
 using cubos::engine::Query;
 
-static void show(Entity entity, Query<Entity, Opt<const Name&>, const ChildOf&, Entity> childOf, Opt<const Name&> name,
-                 tesseratos::EntitySelector& selector)
+bool isSingleWord(const std::string& query)
+{
+    std::istringstream stream(query);
+    std::string word;
+    // TODO: ADD WAY TO CHECK IF ALL THE CHARACTERS IN THE WORD ARE VALID (NOT SPECIAL)
+    stream >> word; // Get the first word
+    return !(stream >> word);
+}
+
+bool isValid(const std::string& query)
+{
+    if (query.empty())
+    {
+        return false;
+    }
+
+    if (isSingleWord(query))
+    {
+        return true;
+    }
+
+    // TODO: ADD Future query validations here
+    return false;
+}
+
+static void showQueries(World& world, Query<Entity, Opt<const Name&>> all, std::string nameQuery,
+                        std::vector<const cubos::core::reflection::Type*> components,
+                        tesseratos::EntitySelector& selector)
+{
+    bool hasComponent = components.empty() ? true : false;
+    auto textColor = (ImVec4)ImColor(255, 255, 255);
+
+    for (auto [entity, name] : all)
+    {
+        for (auto component : components)
+        {
+            if (world.components(entity).has(*component))
+            {
+                hasComponent = true;
+                break;
+            }
```

Comment:
This means that the entity only needs one of the requested components to appear in the result, which is wrong.
It should only appear if all of the components are present in it.
What you want is `if (!has) { hasComponent = false; break; }`, and by default, `hasComponent = true;`